nixos/modules/services/desktops/bonsaid.nix

```
@@ -0,0 +1,167 @@
+          timeout mechanism.
+
+          Only effects transitions with `type = "delay"`.
+        '';
+      };
+      options.event_name = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''
+          Name of the event which should trigger this transition when received by `bonsaid`.
+          Events are sent to `bonsaid` by running `bonsaictl -e <event_name>`.
+
+          Only effects transitions with `type = "event"`.
+        '';
+      };
+      options.transitions = mkOption {
+        type = types.listOf transitionType;
+        default = [ ];
+        description = ''
+          List of transitions out of this state.
+          If left empty, then this state is considered a terminal state and entering it will
+          trigger an immediate transition back to the root state (after processing side effects).
+        '';
+        visible = "shallow";
+      };
+    };
+  cfg = config.services.bonsaid;
+in
+{
+  meta.maintainers = with lib.maintainers; [ colinsane ];
+  meta.buildDocsInSandbox = false;
+
+  options.services.bonsaid = with lib; {
+    enable = mkEnableOption "bonsaid";
+    package = mkPackageOption pkgs "bonsai" { };
+    extraFlags = mkOption {
+      type = types.listOf types.str;
+      default = [ ];
+      description = ''
+        Extra flags to pass to `bonsaid`, such as `[ "-v" ]` to enable verbose logging.
+      '';
+    };
+    settings = mkOption {
+      type = types.listOf transitionType;
+      default = [ ];
+      description = ''
+        State transition definitions. See the upstream [[README]](https://git.sr.ht/~stacyharper/bonsai)
+        for extended documentation and a more complete example.
+      '';
+      example = [
+        {
+          type = "event";
+          event_name = "power_button_pressed";
+          transitions = [
+            {
+              # Hold power button for 600ms to trigger a command
+              type = "delay";
+              delay_duration = 600000000;
+              transitions = [
+                {
+                  type = "exec";
+                  command = [
+                    "swaymsg"
+                    "--"
+                    "output"
+                    "*"
+                    "power"
+                    "off"
+                  ];
+                  # `transitions = []` marks this as a terminal state,
+                  # so bonsai will return to the root state immediately after executing the above command.
+                  transitions = [ ];
+                }
+              ];
+            }
+            {
+              # If the power button is released before the 600ms elapses, return to the root state.
+              type = "event";
+              event_name = "power_button_released";
+              transitions = [ ];
+            }
+          ];
+        }
+      ];
+    };
+    settingsFile = mkOption {
```

Just for consistency with other modules...
```suggestion
    configFile = mkOption {
```